packages/Webkul/Admin/tests/Feature/Catalog/ProductTest.php

```diff
@@ -40,64 +40,699 @@
+    ]);
+});
+
+it('should return the create page of grouped product', function () {
+    // Arrange
+    $product = (new ProductFaker())->getSimpleProductFactory()->create();
+
+    $productId = $product->id + 1;
+
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    postJson(route('admin.catalog.products.store'), [
+        'type'                => 'grouped',
+        'attribute_family_id' => 1,
+        'sku'                 => $sku = fake()->slug(),
+    ])
+        ->assertOk()
+        ->assertJsonPath('data.redirect_url', route('admin.catalog.products.edit', $productId));
+
+    $this->assertDatabaseHas('products', [
+        'type' => 'grouped',
+        'sku'  => $sku,
+        'id'   => $productId,
+    ]);
+});
+
+it('should return the grouped edit page', function () {
+    // Arrange
+    $products = (new ProductFaker())->getSimpleProductFactory()->count(5)->create();
+
+    foreach ($products as $key => $product) {
+        $links['link_' . $key] = [
+            'associated_product_id' => $product->id,
+            'sort_order'            => $key,
+            'qty'                   => rand(10, 100),
+        ];
+    }
+
+    $product = (new ProductFaker())->getSimpleProductFactory()->create([
+        'type' => 'grouped',
+    ]);
+
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    get(route('admin.catalog.products.edit', $product->id), [
+        'sku'               => $product->sku,
+        'url_key'           => $product->url_key,
+        'short_description' => fake()->sentence(),
         'description'       => fake()->paragraph(),
-        'channel'           => core()->getCurrentChannel()->id,
-        'locale'            => core()->getCurrentLocale()->id,
+        'name'              => fake()->words(3, true),
+        'price'             => fake()->randomFloat(2, 1, 1000),
+        'weight'            => fake()->numberBetween(0, 100),
+        'channel'           => core()->getCurrentChannelCode(),
+        'locale'            => app()->getLocale(),
+        'links'             => $links ?? [],
+    ])
+        ->assertOk()
+        ->assertSeeText(trans('admin::app.catalog.products.edit.title'))
+        ->assertSeeText(trans('admin::app.account.edit.back-btn'))
+        ->assertSeeText($product->url_key)
+        ->assertSeeText($product->name)
+        ->assertSeeText($product->short_description)
+        ->assertSeeText($product->description);
+});
+
+it('should update the grouped product', function () {
+    // Arrange
+    $products = (new ProductFaker())->getSimpleProductFactory()->count(5)->create();
+
+    foreach ($products as $key => $product) {
+        $links['link_' . $key] = [
+            'associated_product_id' => $product->id,
+            'sort_order'            => $key,
+            'qty'                   => rand(10, 100),
+        ];
+    }
+
+    $product = (new ProductFaker())->getSimpleProductFactory()->create([
+        'type' => 'grouped',
+    ]);
```

Comment:
Setup group product first,

- add the product factory state
- add the group product factory in faker package

it should be like this,

$product = (new ProductFaker())->getGroupedProductFactory()->create();

Reply:
done